File: assignments/week3/day3/range_sum_BST.py
def rangeSumBST(self, root: TreeNode, L: int, R: int) -> int:
    range_obj = range(L, R + 1)
    sum = 0

    if root == None:
        return 0

    if root.val in range_obj:
        sum += root.val

    if root.right == None:
        return sum + self.rangeSumBST(root.left, L, R)
    if root.left == None:
        return sum + self.rangeSumBST(root.right, L, R)

    return sum + self.rangeSumBST(root.left, L, R) + self.rangeSumBST(
        root.right, L, R)

File: assignments/week3/day3/rangeSumBST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int rangeSumBST(TreeNode* root, int L, int R) {
        int sum = 0;
        queue<TreeNode*> nodes;
        nodes.push(root);
        while(!nodes.empty()){
            auto node = nodes.front();
            nodes.pop();
            if((node->val>=L) && (node->val<=R)){
                sum += node->val;
            }
            if(node->left!=nullptr){
                nodes.push(node->left);
            }
            if(node->right!=nullptr){
                nodes.push(node->right);
            }
        }
        return sum;
    }
};